src/fd.ts

```diff
@@ -116,7 +116,31 @@ export abstract class Fd {
   }
 }
 
+class InoIssuer {
+  // NOTE: ino 0 is reserved for the root directory
+  private next_ino: bigint = 1n;
+
+  issue(_node: unknown): bigint {
```

Comment:
There doesn't seem to be any need to pass in the inode itself.

Reply:
Yes, you are right. I initially tried to add ino recycling mechanism and I thought it requires weakly tracking inode objects. But I changed my mind not to implement this at least for now considering the filesystem is ephemeral.